lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -647,44 +647,38 @@ def retry_multiple(
         if not job_ids:
             raise ValueError("No job_ids provided to retry")
 
-        offending_ids = defaultdict(int)
-        for job_id in job_ids:
-            if job_ids.count(job_id) > 1:
-                offending_ids[job_id] += 1
-
-        if offending_ids.keys():
+        offending_ids = [item for item, count in Counter(job_ids).items() if count > 1]
+        if offending_ids:
             raise ValueError(
                 f"Retry of the same id in the same request is not supported."
-                f" Offending ids:{list(offending_ids.keys())} "
+                f" Offending ids:{offending_ids} "
             )
 
         # Check all inputs before attempting to start submitting jobs
         retried_jobs = []
-        jobs = []
-        batch_jobs = []
         for job_id in job_ids:
+            # Check for presubmission failures
             try:
                 job, batch_job = self._validate_retry_presubmit(
                     job_id=job_id, as_admin=as_admin
                 )
-                jobs.append(job)
-                batch_jobs.append(batch_job)
             except Exception as e:
-                raise RetryFailureException(e)
-
-        # Submit all of the collected jobs
-        for i, job_id in enumerate(job_ids):
+                # Collect the presubmit error and don't submit the job
+                retried_jobs.append({"job_id": job_id, "error": f"{e}"})
+                continue
+            # Presubmit worked, write to the db and submit
```

Comment:
the way this works, IIUC, is that you can have both errors and successful retries in the same response. I understood from the Slack thread that the desired behavior was that there should only be errors or successful retries in the response, not both.

Reply:
This is the desired behavior now. We closed https://github.com/kbase/execution_engine2/pull/414 because that was the behavior we didn't want anymore.

Reply:
That doesn't appear to be the case based on my reading of the thread: https://kbase.slack.com/archives/C01T9GG43C4/p1628286203069900?thread_ts=1628274397.052200&cid=C01T9GG43C4

Reply:
That seems to be the desired behavior to me, and the PR is already approved with this behavior too, so that seems to be the case for @ialarmedalien too

Reply:
The approval is prior to the slack comment I reference above

Reply:
@ialarmedalien Can you comment?

Reply:
This is the desired behaviour: we collect results for every item in the input set, so we could have all successful retries, all errors, or a mixture of both. The important thing is that there is a result for every input item.

Reply:
Ok, it seems that I misinterpreted what AJ said. I still think that implementing partial writes is a mistake and makes it more difficult for the caller, but if both you and AJ want partial writes I'll acquiesce